theano/gpuarray/opt.py

```diff
@@ -1726,6 +1728,66 @@ def local_abstractconv3d_gradinputs_gemm(node):
     return [rval]
 
 
+class LocalCudaMetaOptimizer(LocalMetaOptimizer):
+
+    def time_call(self, fn):
+        start = time.time()
+        fn()[0].sync()
+        return time.time() - start
+
+
+class ConvMetaOptimizer(LocalCudaMetaOptimizer):
+
+    def __init__(self, optimizers=()):
+        super(ConvMetaOptimizer, self).__init__(optimizers)
+
+    def provide_inputs(self, node, inputs):
+        result = {}
+
+        shapes = (node.op.imshp, node.op.kshp)
+        if(node.op.imshp is None or node.op.kshp is None or
+                any([s is None for shape in shapes for s in shape])):
+            return result
+
+        if type(node.op) in [AbstractConv2d, AbstractConv3d]:
+            img, kern = node.inputs
+            for(var, shape) in zip((img, kern), shapes):
+                result[var] = theano.shared(np.random.random(shape).astype(theano.config.floatX),
+                                            var.name, borrow=True)
```

Comment:
And you're missing the broadcast pattern. See https://github.com/Theano/Theano/blob/0.8.X/theano/sandbox/cuda/opt.py#L1715-L1724 for my original code, I guess you can copy this verbatim.